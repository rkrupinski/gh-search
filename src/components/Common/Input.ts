import styled from "styled-components";

export const Input = styled.input`
  box-sizing: border-box;
  height: 2rem;
  padding: 0 0.75rem;
  font-size: 0.875rem;
  line-height: 2;
  border: 0;
  border-radius: ${({ theme }) => theme.layout.borderRadius};
  color: ${({ theme }) => theme.colors.textPrimary};
  background: ${({ theme }) => theme.colors.bgSecondary};
  appearance: none;

  ::placeholder {
    color: ${({ theme }) => theme.colors.textSecondary};
  }
`;
