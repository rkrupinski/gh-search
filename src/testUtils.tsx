import React from "react";
import { render, RenderOptions } from "@testing-library/react";
import { ThemeProvider } from "styled-components";
import { StoreProvider } from "store";
import { theme } from "theme";

const Providers: React.FC = ({ children }) => (
  <StoreProvider>
    <ThemeProvider theme={theme}>{children}</ThemeProvider>
  </StoreProvider>
);

export const renderWithProviders = (
  ui: React.ReactElement,
  options?: Omit<RenderOptions, "queries" | "wrapper">,
) => render(ui, { wrapper: Providers, ...options });

export const noop = () => {
  /**/
};
