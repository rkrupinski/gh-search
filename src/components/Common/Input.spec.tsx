import React from "react";
import { fireEvent } from "@testing-library/react";
import { Input } from "./Input";
import { renderWithProviders, noop } from "testUtils";

describe("Input", () => {
  it("should render correctly", () => {
    const { container, rerender } = renderWithProviders(
      <Input
        type="text"
        autoComplete="off"
        placeholder="Placeholder test"
        value="Value test"
        onChange={noop}
      />,
    );

    expect(container.firstChild).toMatchSnapshot();

    rerender(<Input value="Value test 2" onChange={noop} />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it("should let one handle change", () => {
    const handleChange = jest.fn((e: React.ChangeEvent<HTMLInputElement>) => {
      e.persist();
    });

    const { container } = renderWithProviders(<Input onChange={handleChange} />);

    fireEvent.change(container.firstChild!, {
      target: {
        value: "Test",
      },
    });

    expect(handleChange.mock.calls[0][0].target.value).toBe("Test");
  });
});
