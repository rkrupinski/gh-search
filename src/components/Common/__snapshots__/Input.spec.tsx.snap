// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render correctly 1`] = `
.c0 {
  box-sizing: border-box;
  height: 2rem;
  padding: 0 0.75rem;
  font-size: 0.875rem;
  line-height: 2;
  border: 0;
  border-radius: .5rem;
  color: #000;
  background: #f0f3f4;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.c0::-webkit-input-placeholder {
  color: #777;
}

.c0::-moz-placeholder {
  color: #777;
}

.c0:-ms-input-placeholder {
  color: #777;
}

.c0::placeholder {
  color: #777;
}

<input
  autocomplete="off"
  class="c0"
  placeholder="Placeholder test"
  type="text"
  value="Value test"
/>
`;

exports[`Input should render correctly 2`] = `
<input
  class="sc-AxjAm bGFZxr"
  value="Value test 2"
/>
`;
